### ISSUE TITLE
Cadastro de oportunidade

### ISSUE THREAD

**Issue by lcobucci** [iMastersDev/oportunidades#11]
Após autenticado, o usuário poderá registrar novas oportunidades para que sejam listadas no sistema.
Essencialmente as oportunidades deverão ter título e descrição.

**Comment by netojoaobatista** [comment]
Além do título e descrição, é fundamental que as oportunidades contenham:
1. Título
2. Descrição
3. Local da oportunidade (cidade, etc)
4. Formato do trabalho (in company|home office)
5. Tipo da oportunidade (freela|full time)
6. Tipo de contrato (PJ|CLT)
7. Skils necessárias
8. Faixa salarial

Sim, é possível colocar todas essas informações na descrição, mas se tivermos a separação para local, tipo e skils (utilizando, sei lá, um sistema de tags), podemos em uma versão futura fazer buscas refinadas sem recorrer a full text, notificar usuários por local, skills, etc.

**Comment by lcobucci** [comment]
Claro, por isso o essencialmente hauihaiua coloquei apenas titulo e descrição para que a gente possa burilar e ver o que é REALMENTE necessário na primeira versão da brincadeira.

**Comment by iannsp** [comment]
vejo isso sendo muito simples agora.
os campos estao abordando tudo que precisa, ao que me parece.
o maximo de campos definidos como opções a serem selecionadas é o melhor caminho(IMHO)
a faixa salarial eu indicaria utilizar num futuro algo como um slider, mas agora penso nela como duas informacoes... dois campos como o de...ate

**Comment by suissa** [comment]
Acho interessante criar uma padronização nessas skills, por exemplo deixar um campo para o cliente ir adicionando a skill e ela autocompletar ou sugerir a palavra, para deixar os nomes mais homogeneos. Ficaria mais facil para buscar por tags/skills

**Comment by henriquemoody** [comment]
Concordo com @suissa 

Mas isso é uma questão de modelagem.

**Comment by netojoaobatista** [comment]
Essa questão das skills é um tanto problemático: Se tentarmos fixar as skills, não conseguiremos ter todas cadastradas.

Penso que podemos ir fazendo um "cadastro on-the-fly":
1. Os primeiros usuários poderão adicionar qualquer coisa.
2. Conforme formos tendo skills, vamos fazendo sugestões conforme o cara for digitando.
3. Se o que ele digitou ainda não estiver cadastrado, fazemos a adição na base para sugestão futura.
4. Quando tivermos uma base de skills interessante, fixamos naquelas que já tivermos.

**Comment by alganet** [comment]
Os campos separados são só pra filtrar as vagas?

**Comment by netojoaobatista** [comment]
Sim @alganet, mas esse filtro não existirá agora. Só em uma futura versão.

**Comment by alganet** [comment]
Pô, bacana! Acho que eu até arriscaria uma arquitetura de filtros baseada em folksonomia (tags) sugeridas a partir de um grande campo de texto puro, assim o cara que preparou um TXT com a descrição da vaga pra colar na apinfo e similares não precisaria ficar reformatando e cortando os pedaços pra incluir nos campos separados.

**Comment by netojoaobatista** [comment]
Tenho medo de pesquisas full text.

**Comment by alganet** [comment]
Opa, não é busca full text =) É sugerir tags pra montar uma folksonomia (similar a um tesauro ou taxonomia). É essencialmente extrair palavras relevantes do texto puro que o cara colocar e sugerí-las como tags (permitindo que o cara acrescente outras, se quiser). Mas é só uma idéia, precisamos ver com os usuários o que é melhor e tal.

**Comment by suissa** [comment]
Boa @alganet curti hein! Pois tendo essa arquitetura no futuro quando o cara logar com Linkedin e vc pegar o perfil dele poderá sugerir essas tags daquele texto completo.

**Comment by suissa** [comment]
Tinha que rolar um botao de +1 nos comments =p

**Comment by augustohp** [comment]
Acho que a idéia de **tags** bem produtiva no início. Podemos ir refatorando e acrescentando os campos...
A única coisa que eu separaria mesmo seria o local (estado, cidade), a empresa e o tipo de contratação: PJ (válido pra freelas), CLT.

O resto de informações o cara preenche e boa.

**Comment by diegoholiveira** [comment]
Esse lance de skills é muito bem resolvido no MojoLive, IMO. Acho que poderiamos tentar fazer algo semelhante ao que eles fizeram.

http://img253.imageshack.us/img253/3024/mojoliveskills.png

Podiamos inclusive colocar o lance de qual é a experiência que você procura naquele skill, tipo um combo box com as opções: "experiente, pleno, iniciante". Isso daria a quem posta a oportunidade de colocar skills diferenciadas por produto / tecnologia, por exemplo: "pode ser iniciante em tdd mas experiente em Zend Framework".
Talvez, no futuro, role ate fazer uma certa inteligente pro titulo da oportunidade baseado nesses skills, por exemplo: o cara coloca que conhecer o Respect é muito importante, assim como conhecer Behat. O sistema automaticamente coloca como titulo da vaga: "Desenvolver proficiênte no Respect com conhecimentos de BDD com Behat".

**Comment by ramon-to** [comment]
A questão dos skills é algo muito interessante, principalmente se em uma versão futura do projeto houver a possibilidade de cadastrar as skills dos profissionais, e assim gerar gráficos/relatórios/relação entre o que o mercado necessita e o que "a comunidade" tem a oferecer.

**Comment by suissa** [comment]
+1 @diegoholiveira 
+1 @allmeida

**Comment by thiagophx** [comment]
Gosto da idéia de integração com LinkedIn. Muitas empresas buscam candidatos por lá.
